mtags/src/main/scala/scala/meta/internal/pc/OverrideCompletions.scala

```diff
@@ -0,0 +1,489 @@
+   * @param isCandidate the determination of whether the symbol will be a possible completion item.
+   * @return the list of TextEdit of both method implementations and auto imports.
+   */
+  private def implementAll(
+      typed: Tree,
+      range: l.Range,
+      t: Template,
+      text: String,
+      shouldAddOverrideKwd: Boolean,
+      isCandidate: Symbol => Boolean
+  ): List[l.TextEdit] = {
+    val overrideMembers = getMembers(
+      typed,
+      range,
+      t.pos,
+      text,
+      true,
+      false,
+      isCandidate
+    )
+
+    val allAbstractMembers = overrideMembers
+      .filter(_.sym.isAbstract)
+
+    val (allAbstractEdits, allAbstractImports) = toEdits(allAbstractMembers)
+
+    if (allAbstractEdits.length > 0) {
+
+      // infer necessary indent
+      //
+      // |object Test {
+      // |    class Foo extends Bar {} // inferred to 4
+      // |}
+      val lineStart = t.pos.source.lineToOffset(t.pos.line - 1)
+      val necessaryIndent = inferIndent(lineStart, text)
+
+      // infer indent for implementations
+      // if there's declaration in the class/object, follow its indent.
+      // otherwise the indent default to 2
+      val indent = typed.tpe.decls
+        .filter(sym =>
+          !sym.isSynthetic &&
+            !sym.isPrimaryConstructor &&
+            sym.pos.line != t.pos.line // filter out explicit primary constructor `class Foo(x: Int) ...`
+        )
+        .headOption
+        .map(existing => {
+          " " * inferIndent(
+            t.pos.source.lineToOffset(existing.pos.line - 1),
+            text
+          )
+        })
+        .getOrElse {
+          " " * (necessaryIndent + DefaultIndent)
+        }
+
+      val shouldCompleteBraces = !hasBody(text, t)
+
+      // if the both opening/closing braces located in a line:
+      // ```
+      // object {
+      //   class Foo extends Bar {}
+      // }
+      // ```
+      // or there's no body like this `class Foo extends Bar`.
+      // Add an newline and indent in the end of implementations, so that
+      // the closing brace is indented.
+      //
+      // object {
+      //   class Foo extends Bar {
+      //     override def foo = ???
+      //   }
+      // }
+      val lastIndent =
+        if (t.pos.source.offsetToLine(t.pos.start) ==
+            t.pos.source.offsetToLine(t.pos.end) || shouldCompleteBraces)
+          "\n" + " " * necessaryIndent
+        else ""
+
+      // Add opening/closing braces
+      // `object Foo extends Bar` to
+      // ```
+      // object Foo extends Bar {
+      //   override def method: Int = ???
+      // }
+      // ```
+      val start = if (shouldCompleteBraces) s" {\n${indent}" else s"\n${indent}"
+      val end = if (shouldCompleteBraces) s"${lastIndent}}" else lastIndent
+      val implementAll = new l.TextEdit(
+        range,
+        allAbstractEdits
+          .map(_.getNewText)
+          .mkString(
+            start,
+            s"\n${indent}",
```

Comment:
Let's maybe add another newline between members?

It's usually what people would want I think.

So instead of:
```
class A{
  def a = ???
  def b = ???
}
```
we would have:
```
class A{
  def a = ???
  
  def b = ???
}
```

I would also consider maybe newline at the start or end.

Reply:
Personally, I prefer the dense result (without empty lines), but meanwhile, the intellij-scala-plugin seems to add empty lines between methods, and it's probably better to add empty lines as you suggested 👍 so that users from IntelliJ can easily get used to the metals.

Reply:
So one thing to think about with this is now we have the "completion" version giving you the condensed version with no spaces and the "code action" version giving extra spaces. I sort of think we should be consistent.

Reply:
> I sort of think we should be consistent.

Ah, that's correct, I'm still leaning toward add spaces for both codeaction and completions.
What do you think about this ?

Reply:
For reference, added spaces between abstract members for completion-version https://github.com/scalameta/metals/pull/1379/commits/e100253dce0a50a1201510f9bec11fbe8f298bbb :)
We can revert it if we don't need it.

Reply:
I don't have a strong preference on it. I just thing whichever one we choose we should do it the same in both. I'd say leave them they way you have them 👍